### PR TITLE
PLENTY-429_plentyShop_5_0_47_compatibility

### PR DESCRIPTION
**Pull request by dblaichinger**
PLENTY-429_plentyShop_5_0_47_compatibility

### CODE REVIEW

**Review by TobiasGraml11 — APPROVED** [review]

File: src/Services/SearchService.php [hunk 1]
```diff
@@ -465,7 +483,7 @@ private function getVariationIdForRedirect(string $query, array $documents)
         foreach ($documents as $document) {
             $variation = $document['data']['variation'];
             $barcodes = $document['data']['barcodes'] ?? [];
-            $variationPrice = $this->getCheapestPrice($document['data']['salesPrices']);
+            $variationPrice = $document['data']['prices']['default']['price']['value'] ?: 0;
```
Comment (TheKeymaster) [inline src/Services/SearchService.php:486]: Is this actually the cheapest price?

File: src/Services/SearchService.php [hunk 2]
```diff
@@ -410,26 +418,36 @@ private function getProductDetailUrl(Response $response)
             $productId = explode('_', $productId)[0];
         }
 
-        $result = $itemSearchService->getResult(
-            $this->getSearchFactory()->hasItemId($productId)
-        );
+        // Necessary to retrieve variation data and prices
+        $resultFields = $this->loadResultFieldTemplate(ResultFieldTemplate::TEMPLATE_LIST_ITEM);
```
Comment (TheKeymaster) [inline src/Services/SearchService.php:422]: How do you find this? Did you check the Ceres code for this?

File: tests/Services/SearchServiceTest.php [hunk 3]
```diff
@@ -192,27 +193,52 @@ public function testItemVariantIdExtractingForRedirectUrlGeneration()
...
+                'total' => 1,
+                'documents' => $this->getMultipleItemsDocuments(
+                    [
+                            '0' => [
+                                'id' => $mainVariationId,
+                                'price' => 10.00,
+                                'isMain' => true
+                            ]
+                        ]
+                )
```
Comment (TheKeymaster) [inline tests/Services/SearchServiceTest.php:233]: Please check the indentation.

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

File: src/Services/SearchService.php (hunk 1 above)
Comment (dblaichinger) [reply src/Services/SearchService.php:486]: Note: Also check why unit test isn't failing.

File: src/Services/SearchService.php (hunk 2 above)
Comment (dblaichinger) [reply src/Services/SearchService.php:422]: Discussed in call

File: src/Services/SearchService.php (hunk 1 above)
Comment (dblaichinger) [reply src/Services/SearchService.php:486]: > Is this actually the cheapest price?

Yes, the cheapeast price gets set on line 518, so the logic works as before, just without need to parse the `salesPrices` array.

**Review by TheKeymaster — APPROVED** [review]

**Pull request merged**